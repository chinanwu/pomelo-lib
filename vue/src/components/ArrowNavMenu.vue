<template>
  <ul
    class="ArrowNavMenu"
    role="menu"
    :aria-activedescendant="activeDescendant"
  >
    <li
      v-for="(item, index) in items"
      :key="'aNM-' + index"
      role="menuitem"
      :id="'aNM-' + index + '-' + item"
    >
      <button
        :id="'aNM-btn-' + index"
        class="ArrowNavMenu__item"
        :class="{ 'ArrowNavMenu__item--selected': selected === index }"
        @click="handleClick(index)"
        @keydown="handleKeyDown"
      >
        {{ item }}
      </button>
    </li>
  </ul>
</template>

<script>
// TODO: role menu vs menubar - Dynamic setting via window size maybe?

export default {
  name: "ArrowNavMenu",
  props: {
    items: {
      type: Array,
      required: true,
      validator: values =>
        values.every(
          value => typeof value === "string" || value instanceof String
        )
    },
    active: {
      type: String,
      default: null // will be computed as first item
    }
  },
  data() {
    return {
      selected: this.active ? this.items.indexOf(this.active) : 0
    };
  },
  computed: {
    activeDescendant() {
      return "aNM-" + this.selected + "-" + this.items[this.selected];
    }
  },
  methods: {
    setSelected(i) {
      this.$emit("selected", this.items[i]);
      this.selected = i;
    },
    handleClick(i) {
      this.setSelected(i);
    },
    handleKeyDown(event) {
      if (
        event &&
        !event.shiftKey &&
        !event.ctrlKey &&
        !event.altKey &&
        !event.metaKey &&
        event.key
      ) {
        // https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-actions-active-descendant.html
        if (event.key === "ArrowDown" || event.key === "ArrowRight") {
          event.preventDefault();
          this.selected === this.items.length - 1
            ? this.setSelected(0)
            : this.setSelected(this.selected + 1);
        } else if (event.key === "ArrowUp" || event.key === "ArrowLeft") {
          event.preventDefault();
          this.selected === 0
            ? this.setSelected(this.items.length - 1)
            : this.setSelected(this.selected - 1);
        } else if (event.key === "Home") {
          event.preventDefault();
          this.setSelected(0);
        } else if (event.key === "End") {
          event.preventDefault();
          this.setSelected(this.items.length - 1);
        } else {
          if (/^[a-zA-Z]$/.test(event.key)) {
            // Note from Salmon: This is seriously so cool y'all!
            for (let i = 1; i < this.items.length; i++) {
              // Start at one so it doesn't count itself as an option to jump to
              const temp = this.selected + i;
              if (temp >= this.items.length) {
                const firstLetter = this.items[temp % this.items.length].charAt(
                  0
                );
                if (
                  firstLetter === event.key ||
                  firstLetter === event.key.toLowerCase() ||
                  firstLetter === event.key.toUpperCase()
                ) {
                  this.setSelected(temp % this.items.length);
                  return;
                }
              } else {
                const firstLetter = this.items[temp].charAt(0);
                if (
                  firstLetter === event.key ||
                  firstLetter === event.key.toLowerCase() ||
                  firstLetter === event.key.toUpperCase()
                ) {
                  this.setSelected(temp);
                  return;
                }
              }
            }
          }
        }
      }
    }
  },
  emits: ["selected"]
};
</script>

<style lang="less" scoped>
.ArrowNavMenu {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: space-evenly;
  font: inherit;
}

.ArrowNavMenu__item {
  color: black;
  border: none;
  background-color: inherit;
  font-size: 3.6rem;
  margin: 1rem;
  cursor: pointer;
  transition: all 0.2s;

  &:focus,
  &:hover {
    text-decoration: underline;
  }
}

.ArrowNavMenu__item--selected {
  color: #3400fe;

  &::before {
    content: ">";
    color: #de0202;
    display: inline-block;
    animation: fadeIn 0.3s;
    padding: 0 1rem;
    border: none;
  }
}

@media (prefers-reduced-motion) {
  .ArrowNavMenu__item--selected::before {
    animation: none;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
    transform: translateX(-1rem);
  }
  to {
    opacity: 1;
    transform: translateX(0);
  }
}

// iPad Pro 11"
@media only screen and (max-width: 834px) {
  .ArrowNavMenu__item {
    font-size: 2.4rem;
  }
}

// iPhone 11 Pro Max, iPhone 8
@media only screen and (max-width: 414px) {
  .ArrowNavMenu {
    flex-direction: column;
  }

  .ArrowNavMenu__item {
    font-size: 3.6rem;
  }
}

// iPhone SE
@media only screen and (max-width: 320px) {
  .ArrowNavMenu__item {
    font-size: 2.4rem;
  }
}
</style>
