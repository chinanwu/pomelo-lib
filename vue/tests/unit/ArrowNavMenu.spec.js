import { shallowMount } from "@vue/test-utils";

import ArrowNavMenu from "@/components/ArrowNavMenu";

describe("ArrowNavMenu", () => {
  describe("rendering", () => {
    it("renders ArrowNavMenu", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
    });

    it("renders ArrowNavMenu with non default active", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"],
          active: "Two"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe("behaviour", () => {
    it("sets selected on click", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"],
          active: "Two"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      wrapper.find("#aNM-btn-0").trigger("click");
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["One"]]);
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
    });

    it("selects next item with arrow down", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowDown",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["Two"]]);
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects next item with arrow right", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"],
          active: "Two"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");

      wrapper.find("#aNM-btn-1").trigger("keydown", {
        key: "ArrowRight",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["One"]]);
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects previous item with arrow up", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"],
          active: "Two"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");

      wrapper.find("#aNM-btn-1").trigger("keydown", {
        key: "ArrowUp",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["One"]]);
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects previous item with arrow left", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowLeft",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["Two"]]);
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects first item in items with home", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"],
          active: "Two"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");

      wrapper.find("#aNM-btn-1").trigger("keydown", {
        key: "Home",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["One"]]);
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects last item in items with end", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "End",
        preventDefault
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["Two"]]);
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-Two");
      expect(preventDefault).toHaveBeenCalledTimes(1);
    });

    it("selects next item with letter starting with key pressed", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["this", "That", "There"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-this");

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "t"
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["That"]]);
      expect(wrapper.vm.selected).toBe(1);
      expect(wrapper.vm.activeDescendant).toBe("aNM-1-That");
    });

    it("selects next item with letter starting with key pressed circular", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["this", "That", "There"],
          active: "There"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(2);
      expect(wrapper.vm.activeDescendant).toBe("aNM-2-There");

      wrapper.find("#aNM-btn-2").trigger("keydown", {
        key: "T"
      });
      expect(wrapper.emitted()).toHaveProperty("selected");
      expect(wrapper.emitted().selected).toEqual([["this"]]);
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-this");
    });

    it("does nothing if extra keys are pressed", () => {
      const preventDefault = jest.fn();
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowLeft",
        shiftKey: true,
        preventDefault
      });
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).not.toHaveBeenCalled();

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowLeft",
        ctrlKey: true,
        preventDefault
      });
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).not.toHaveBeenCalled();

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowLeft",
        altKey: true,
        preventDefault
      });
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).not.toHaveBeenCalled();

      wrapper.find("#aNM-btn-0").trigger("keydown", {
        key: "ArrowLeft",
        metaKey: true,
        preventDefault
      });
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
      expect(preventDefault).not.toHaveBeenCalled();
    });

    it("does nothing if invalid event", () => {
      const wrapper = shallowMount(ArrowNavMenu, {
        props: {
          items: ["One", "Two"]
        }
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown");
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");

      wrapper.find("#aNM-btn-0").trigger("keydown", {});
      expect(wrapper.vm.selected).toBe(0);
      expect(wrapper.vm.activeDescendant).toBe("aNM-0-One");
    });
  });

  describe("prop validation", () => {
    it("throws error if not valid items array", () => {
      const valid = ["test", "here", "loooong"];
      const invalid = [0];
      const validator = ArrowNavMenu.props.items.validator;
      expect(validator(valid)).toBe(true);
      expect(validator(invalid)).toBe(false);
    });
  });
});
