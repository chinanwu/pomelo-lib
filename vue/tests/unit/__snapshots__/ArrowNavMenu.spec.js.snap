// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowNavMenu behaviour does nothing if extra keys are pressed 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour does nothing if invalid event 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects first item in items with home 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects item with letter starting with key pressed 1`] = `
<ul
  aria-activedescendant="aNM-0-this"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-this"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      this
    </button>
  </li>
  <li
    id="aNM-1-That"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      That
    </button>
  </li>
  <li
    id="aNM-2-There"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-2"
    >
      There
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects last item in items with end 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects next item with arrow down 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects next item with arrow down or arrow right 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects next item with arrow right 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects next item with letter starting with key pressed 1`] = `
<ul
  aria-activedescendant="aNM-0-this"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-this"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      this
    </button>
  </li>
  <li
    id="aNM-1-That"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      That
    </button>
  </li>
  <li
    id="aNM-2-There"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-2"
    >
      There
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects next item with letter starting with key pressed circular 1`] = `
<ul
  aria-activedescendant="aNM-2-There"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-this"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      this
    </button>
  </li>
  <li
    id="aNM-1-That"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      That
    </button>
  </li>
  <li
    id="aNM-2-There"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-2"
    >
      There
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects previous item with arrow left 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects previous item with arrow up 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour selects previous item with arrow up or arrow left 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu behaviour sets selected on click 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu rendering renders ArrowNavMenu 1`] = `
<ul
  aria-activedescendant="aNM-0-One"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;

exports[`ArrowNavMenu rendering renders ArrowNavMenu with non default active 1`] = `
<ul
  aria-activedescendant="aNM-1-Two"
  class="ArrowNavMenu"
  role="menu"
>
  
  <li
    id="aNM-0-One"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item"
      id="aNM-btn-0"
    >
      One
    </button>
  </li>
  <li
    id="aNM-1-Two"
    role="menuitem"
  >
    <button
      class="ArrowNavMenu__item ArrowNavMenu__item--selected"
      id="aNM-btn-1"
    >
      Two
    </button>
  </li>
  
</ul>
`;
